### PR TITLE
chore: i18n setup and translate some pages

### PR DESCRIPTION
**Pull request by flagrede**
## Description

Setup lingui as the 18n library
- install dependencies and vite plugin
- add script to extract messages
- setup i18n provider for all clients
- setup base config with en and es locales
- setup eslint plugin
- load default locale on app loading
- add some documentation on usage

Translate a few pages as examples:
- Additionality
- BasicInfo
- BatchDetails
- BasketDetails
- Buyers
- Certificate
- ChooseCreditClass
- ConnectWalletPage
- CreateBatch

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### How to test

1.

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: web-marketplace/lingui.config.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import type { LinguiConfig } from '@lingui/conf';
```
Comment (blushi) [inline web-marketplace/lingui.config.ts:1]: could the config be similar for web-components? should we have one root config file and then config files in each packages extended from the root one? ref: https://lingui.dev/guides/monorepo
The thing is that web-components is also used by web-www that we doesn't need to be multilingual (at least for now, as part of this project)

File: web-marketplace/package.json [hunk 2]
```diff
@@ -90,7 +92,8 @@
     "graphql:codegen": "graphql-codegen --config codegen.yml",
     "graphql:codegen-sanity": "graphql-codegen --config sanity-codegen.yml",
     "graphql:codegen-indexer": "graphql-codegen --config indexer-codegen.yml",
-    "analyze": "source-map-explorer 'build/static/js/*.js'"
+    "analyze": "source-map-explorer 'build/static/js/*.js'",
+    "messages:extract": "lingui extract"
```
Comment (blushi) [inline web-marketplace/package.json:96]: should we integrate `lingui compile` as part of our build process?

**Review by blushi — COMMENTED** [review]
Looking good, just a couple questions on how to handle monorepo and building

File: web-marketplace/lingui.config.ts (hunk 1 above)
Comment (flagrede) [reply web-marketplace/lingui.config.ts:1]: Sure, I'll update that.

File: web-marketplace/package.json (hunk 2 above)
Comment (flagrede) [reply web-marketplace/package.json:96]: The compile step is already handled by the Vite plugin.

Comment (blushi) [reply web-marketplace/package.json:96]: ok nice!

File: web-marketplace/lingui.config.ts (hunk 1 above)
Comment (flagrede) [reply web-marketplace/lingui.config.ts:1]: I did not add a `lingui.config.ts` to `web-components` though, as this package should probably not be concerned with i18n. If there are hard-coded strings in `web-components`, we might want to extract them.

Comment (blushi) [reply web-marketplace/lingui.config.ts:1]: ok yeah makes sense, we do have some hard coded strings in there
so the idea would be to have them set as props instead?

Comment (flagrede) [reply web-marketplace/lingui.config.ts:1]: If this is too much of a refactor for the scope of the project, we can reconsider, but ideally, `web components' should provide language-agnostic components.

Comment (flagrede) [reply web-marketplace/lingui.config.ts:1]: Yes, the idea would be to send these strings as props.

**Review by blushi — APPROVED** [review]
LGTM, just a small nit

File: README.md [hunk 3]
```diff
@@ -174,6 +177,22 @@ If you are using VsCode, there are suggested workspace settings in `.vscode/sett
...
+- attributes: \_(msg\`static text\`)
+- constants:
+  - const elements = [msg\`some element\`]
+  - \_(elements[0])
+
+### Important notes
+
+`<Trans>` component should be imported from `@lingui/macro` most of the time. This component is used at compile-time.
+There is another `<Trans>` component in `@lingui/react` for [special cases](https://lingui.dev/tutorials/react-patterns#lazy-translations). This component is used at runtime.
+
```
Comment (blushi) [inline README.md:195]: could we add a note about `i18n:extract` new command in here?

**Pull request merged**